### PR TITLE
[superseded] add math.copySign

### PR DESCRIPTION
**Pull request by ringabout**
ref nim-lang/RFCs#92

C99 function
http://www.cplusplus.com/reference/cmath/copysign/

Python:
https://docs.python.org/3/library/math.html#math.copysign

Crystal:
https://github.com/crystal-lang/crystal/blob/5999ae29beacf4cfd54e232ca83c1a46b79f26a5/src/math/math.cr#L95

Dlang:
https://dlang.org/library/std/math/copysign.html

Rust:
https://doc.rust-lang.org/beta/std/primitive.f64.html#method.copysign

### CODE REVIEW

**Comment by ringabout** [comment]
`sgn` is ugly
https://stackoverflow.com/questions/1986152/why-doesnt-python-have-a-sign-function
And `copysign` is more useful
https://bugs.python.org/msg59154

File: lib/pure/math.nim [hunk 1]
```diff
@@ -914,6 +916,23 @@ when not defined(js):
...
+      doAssert copySign(10'f32, -1.0) == -10.0
+
+      doAssert copySign(Inf, -1.0) == -Inf
+      doAssert copySign(-Inf, 1.0) == Inf
+      doAssert copySign(1.0, -0.0) == -1.0
+      doAssert copySign(0.0, -0.0) == -0.0
+      doAssert copySign(-1.0, 0.0) == 1.0
+      doAssert copySign(10.0, 0.0) == 10.0
+
+    # TODO add examples for isNaN and JS version
```
Comment (timotheecour) [inline lib/pure/math.nim:934]: and VM; which is why might be preferable to have auxiliary importc proc, as was for with `math.isNaN` see https://github.com/nim-lang/Nim/pull/16179

File: lib/pure/math.nim [hunk 2]
```diff
@@ -914,6 +916,23 @@ when not defined(js):
       ##  echo log2(0.0)  # -inf
       ##  echo log2(-2.0) # nan
 
+  proc copySign*[T: float32 | float64](x, y: T): T {.importc: "copysign", header: "math.h".} =
```
Comment (timotheecour) [inline lib/pure/math.nim:919]: T: SomeFloat

**Comment by timotheecour** [comment]
superseded by #16406

**Pull request closed without merging**